packages/api/src/controllers/task.js

```diff
@@ -0,0 +1,180 @@
+  createdAt: `task.data->'createdAt'`,
+  userId: `task.data->>'userId'`,
+  "user.email": { val: `users.data->>'email'`, type: "full-text" },
+  type: `task.data->>'type'`,
+};
+
+app.get("/", authMiddleware({}), async (req, res) => {
+  let { limit, cursor, all, event, allUsers, order, filters, count } =
+    req.query;
+  if (isNaN(parseInt(limit))) {
+    limit = undefined;
+  }
+
+  if (req.user.admin && allUsers && allUsers !== "false") {
+    const query = parseFilters(fieldsMap, filters);
+    if (!all || all === "false") {
+      query.push(sql`task.data->>'deleted' IS NULL`);
+    }
+
+    let fields =
+      " task.id as id, task.data as data, users.id as usersId, users.data as usersdata";
+    if (count) {
+      fields = fields + ", count(*) OVER() AS count";
+    }
+    const from = `task left join users on task.data->>'userId' = users.id`;
+    const [output, newCursor] = await db.task.find(query, {
+      limit,
+      cursor,
+      fields,
+      from,
+      order: parseOrder(fieldsMap, order),
+      process: ({ data, usersdata, count: c }) => {
+        if (count) {
+          res.set("X-Total-Count", c);
+        }
+        return { ...data, user: db.user.cleanWriteOnlyResponse(usersdata) };
+      },
+    });
+
+    res.status(200);
+
+    if (output.length > 0 && newCursor) {
+      res.links({ next: makeNextHREF(req, newCursor) });
+    }
+    return res.json(output);
+  }
+
+  const query = parseFilters(fieldsMap, filters);
+  query.push(sql`task.data->>'userId' = ${req.user.id}`);
+
+  if (!all || all === "false") {
+    query.push(sql`task.data->>'deleted' IS NULL`);
+  }
+
+  let fields = " task.id as id, task.data as data";
+  if (count) {
+    fields = fields + ", count(*) OVER() AS count";
+  }
+  const from = `task`;
+  const [output, newCursor] = await db.task.find(query, {
+    limit,
+    cursor,
+    fields,
+    from,
+    order: parseOrder(fieldsMap, order),
+    process: ({ data, count: c }) => {
+      if (count) {
+        res.set("X-Total-Count", c);
+      }
+      return { ...data };
+    },
+  });
+
+  res.status(200);
+
+  if (output.length > 0) {
+    res.links({ next: makeNextHREF(req, newCursor) });
+  }
+
+  return res.json(output);
+});
+
+app.get("/:id", authMiddleware({}), async (req, res) => {
+  const os = await req.store.get(`task/${req.params.id}`);
+  if (!os) {
+    res.status(404);
+    return res.json({
+      errors: ["not found"],
+    });
+  }
+
+  if (req.user.admin !== true && req.user.id !== os.userId) {
+    res.status(403);
+    return res.json({
+      errors: ["user can only request information on their own tasks"],
+    });
+  }
+
+  res.json(os);
+});
+
+app.post("/", authMiddleware({}), validatePost("task"), async (req, res) => {
+  const id = uuid();
+  const doc = validateTaskPayload(id, req.user.id, Date.now(), req.body);
+  try {
+    await db.task.create(doc);
+  } catch (e) {
+    console.error(e);
+    throw e;
+  }
```

Comment:
Shouldn't need the try/catch, will just get picked up by our middleware

```suggestion
  await db.task.create(doc);
```